app/uk/gov/hmrc/helptosave/actors/UCThresholdManager.scala

```diff
@@ -56,104 +62,206 @@ class UCThresholdManager(thresholdConnectorProxyActor: ActorRef,
-      maybeRequester match {
-        case None ⇒
-          result.fold ({
-            e ⇒
-              pagerDutyAlerting.alert ("Could not obtain initial UC threshold value from DES")
-              //If Des is down, retry
-              getDESRetries.retry (()).fold (
-                logger.warn (s"Could not obtain initial UC threshold value from DES: $e. Job to retry getting value from DES " +
-                  "already scheduled - no new job scheduled")
-              ) {
-                  t ⇒
-                    logger.warn (s"Could not obtain initial UC threshold value from DES: $e. Job to retry getting value from DES " +
-                      s"scheduled to run in ${
-                        Time.nanosToPrettyString (t.toNanos)
-                      }")
-                }
-          }, {
-            value ⇒
-              logger.info (s"Successfully obtained the UC threshold value $value from DES")
-              context become ready (value)
-          })
-
-        case Some(requester) ⇒
-          result.fold ({
-            e ⇒
-              logger.warn(s"Could not obtain UC threshold value from DES: $e ")
-              requester ! GetThresholdValueResponse(None)
-          }, {
-            value ⇒
-              logger.info (s"Successfully obtained the UC threshold value $value from DES")
-              requester ! GetThresholdValueResponse(Some(value))
-              getDESRetries.cancelAndReset()
-              context become ready (value)
-          })
-      }
+    case r: UCThresholdManager.GetDESThresholdValueResponse ⇒
+      handleDESThresholdValueFromNotReady(r)
+
+    case UpdateWindow ⇒
+      logger.warn("[notReady] Time to start updating threshold value but in notReady state - no action will be taken")
 
   }
 
   def ready(thresholdValue: Double): Receive = {
 
+    case GetDESThresholdValue ⇒
+      logger.warn("[ready] Received unexpected message: GetDESThresholdValue")
+
     case GetThresholdValue ⇒
       sender() ! GetThresholdValueResponse(Some(thresholdValue))
 
-    case UCThresholdManager.GetDESThresholdValueResponse(maybeRequester, result) ⇒
-      val value =
-        result.fold(
-          {
-            e ⇒
-              logger.warn (s"Call to get UC threshold value from DES failed: $e. But we already have an initial value from DES")
-              thresholdValue
-          }, {
-            value ⇒
-              logger.info (s"Successfully obtained the UC threshold value $value from DES")
-              value
-          }
-        )
+    case r: UCThresholdManager.GetDESThresholdValueResponse ⇒
+      handleDESThresholdValueFromReady(r, thresholdValue)
+
+    case UpdateWindow ⇒
+      logger.info("[ready] Time to start updating threshold value - proceeding to updating state")
+      updateThresholdValueJob = Some(scheduleEndOfUpdateWindow())
+      context become inUpdateWindow(endOfWindowTriggered = false)
+  }
+
+  def inUpdateWindow(endOfWindowTriggered: Boolean): Receive = {
```

Comment:
with these changes the `UCThresholdManager ` actor has too much logic in it and became complex.
why don't we move the new schedule logic out of this to keep it clean  because this actor shouldn't worry about `inUpdateWindow` but only about getting values from DES